Generate.py

```diff
@@ -339,6 +332,20 @@ def update_weights(weights: dict, new_weights: dict, update_type: str, name: str
                 raise Exception(f"Cannot apply merge to non-dict, set, or list type {option_name},"
                                 f" received {type(new_value).__name__}.")
             cleaned_weights[option_name] = cleaned_value
+        elif option.startswith("-") and option_name in weights:
```

Comment:
I notice that this method of stripping the `+-` and only looking at the first character means that for option `my_option`, the following keys would all work:

| key | effect |
| --- | --- |
| +my_option | appends |
| -my_option | removes |
| - my_option | replaces |
| ------my_option | removes |
| +-+-+-my_option | appends |
| -+++++++++++my_option | removes |

Reply:
yes